runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -1821,6 +1843,71 @@ int32_t chpl_comm_getMaxThreads(void) {
 }
 
 
+////////////////////////////////////////
+//
+// Interface: barriers
+//
+
+//
+// We do a simple tree-based split-phase barrier, with locale 0 as the
+// root of the tree.  Each of the locales has a barrier_info_t struct,
+// and knows the address of that struct in its child locales (locales
+// num_children*my_idx+1 - num_children*my_idx+num_children) and its
+// parent (locale (my_idx-1)/num_children).  Notify and release flags on
+// all locales start out 0.  The notify step consists of each locale
+// waiting for its children, if it has any, to set the child_notify
+// flags in its own barrier info struct to 1, and then if it is not
+// locale 0, setting the child_notify flag corresponding to itself in
+// its parent's barrier info struct to 1.  Thus notification propagates
+// up from the leaves of the tree to the root.  In the wait phase each
+// locale except locale 0 waits for the parent_release flag in its own
+// barrier info struct to become 1.  Once a locale sees that, it clears
+// all of the flags in its own struct and then sets the parent_release
+// flags in both of its existing children to 1.  Thus releases propagate
+// down from locale 0 to the leaves.  Once waiting is complete at the
+// leaves, all of the flags throughout the job are back to 0 and the
+// process can repeat.
+//
+// Note that we can (and do) do other things while waiting for notify
+// and release flags to be set.  In fact we have to task-yield while
+// doing so, in case the PUTs need to be done via AM for some reason
+// (unregistered memory, e.g.).
+//
+// TODO: vectorize the child PUTs.
+//
+#define BAR_TREE_NUM_CHILDREN 64
```

Comment:
I think we should go with 2 for now (which is what we used in ugni before switching chained/vector puts in https://github.com/chapel-lang/chapel/pull/8557)

Reply:
I'm not sure it matters.  We could even go with 1.  Is 2 preferable for some reason?  I went with 64 just because it might be one fewer thing to change in the future.

Reply:
I would imagine 2 will be better for performance while we're not doing chained puts, but you're probably right that it doesn't really matter